### PR TITLE
Copy constructor for Toml class

### PR DESCRIPTION
**Pull request by Spartee**
This PR adds a copy constructor for the Toml library. This is very useful in the creation of .lock files and therefore is needed in the TOML library for Mason. Also added is a corner case for arrays that should have been included in the first pr. 

@ben-albrecht

### CODE REVIEW

File: test/modules/packages/Toml/TOML.chpl [hunk 1]
```diff
@@ -434,6 +434,20 @@ Parser module with the Toml class for the Chapel TOML library.
        this.tag = fieldArr;
      }
 
+     // Clone
+     proc init(root: Toml) {
```
Comment (ben-albrecht) [inline test/modules/packages/Toml/TOML.chpl]: I think this deserves a test

Comment (Spartee) [inline test/modules/packages/Toml/TOML.chpl]: On it
